storage/iam.rb

```diff
@@ -62,14 +68,56 @@ def remove_bucket_iam_member project_id:, bucket_name:, role:, member:
   storage = Google::Cloud::Storage.new project_id: project_id
   bucket = storage.bucket bucket_name
 
-  bucket.policy do |policy|
-    policy.remove role, member
+  bucket.policy requested_policy_version: 3 do |policy|
+    policy.bindings.each do |binding|
+      if binding.role == role
+        if binding.members.count > 1
+          # At least 2 members exist in the binding
+          binding.members.delete member
+        else
+          # Last member in the list, delete binding from policy.
+          policy.bindings.remove binding
```

Comment:
Is it valid to remove while iterating over the bindings list?

Reply:
:man_facepalming: good catch will fix.